drivers/interrupt_controller/intc_esp32c3.c

```diff
@@ -236,6 +251,13 @@ int esp_intr_enable(int source)
 	uint32_t key = irq_lock();
 	uint32_t irq = esp_intr_find_irq_for_source(source);
 
+#if defined(CONFIG_SOC_SERIES_ESP32C6)
```

Comment:
Why only for C6?

Reply:
IRQ_NA was introduced as a safety-only check when IRQ allocation was implemented for C6.
As C3 (and C2) use a different implementation for `esp_intr_find_irq_for_source()` it never returns IRQ_NA.